### PR TITLE
2734-Integrate-spec-200

### LINKED ISSUE #2734: Integrate spec 2.0.0
A first (not definitive) milestone.
Changeslog in: https://github.com/pharo-spec/Spec/releases/tag/v2.0.0-alpha

### PR DESCRIPTION
**Pull request by guillep**
Fix #2734: Integrate latest changes to spec + tons of tests

### CODE REVIEW

File: src/BaselineOfSpec/BaselineOfSpec.class.st [hunk 1]
```diff
@@ -1,38 +1,72 @@
...
-		spec package: 'Spec-MorphicAdapters'.
-		spec package: 'Spec-PolyWidgets'.
-		spec package: 'Spec-Tools'.
-		spec package: 'Spec-StubAdapter'.
-		spec package: 'Spec-Deprecated'.
-	].
+
+	spec for: #common do: [ 
+		spec 
+			preLoadDoIt: #addHackToLoadSpec;
```
Comment (jecisc) [inline src/BaselineOfSpec/BaselineOfSpec.class.st]: The hacks in baseline should be removed. They are here to update Spec.

File: src/BaselineOfSpec/MCVersionLoader.extension.st [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Extension { #name : #MCVersionLoader }
+
+{ #category : #'*BaselineOfSpec' }
```
Comment (jecisc) [inline src/BaselineOfSpec/MCVersionLoader.extension.st]: This should be removed for the integration in Pharo. It's a hack to update Spec.

**Review by jecisc — CHANGES_REQUESTED** [review]

File: src/BaselineOfSpec/BaselineOfSpec.class.st (hunk 1 above)
Comment (guillep) [reply src/BaselineOfSpec/BaselineOfSpec.class.st]: Done + https://github.com/pharo-spec/Spec/issues/337

File: src/BaselineOfSpec/MCVersionLoader.extension.st (hunk 2 above)
Comment (guillep) [reply src/BaselineOfSpec/MCVersionLoader.extension.st]: Done + [pharo-spec/Spec#337](https://github.com/pharo-spec/Spec/issues/337)

**Comment by guillep** [comment]
The problem seems to be that the new Spec version uses a stateful trait 

```smalltalk
Trait named: #TDynamicPresenter
	 uses: {}
	 slots: { #presenters }
	 category: 'Spec-Core-Base'
```

But we are not being able to export it to mcz during bootstrap.

**Comment by Ducasse** [comment]
Tx guille!!!!

**Comment by Ducasse** [comment]
You have a dependencies problem.

**Comment by Ducasse** [comment]
Loaded -> GT-Inspector-tonel.1 --- tonel:///builds/workspace/uest_and_branch_Pipeline_PR-2735/32/src --- cacheError: This package depends on the following classes:
[32]   WindowModel
[32]   ComposableModel
[32] You must resolve these dependencies before you will be able to load these definitions: 
[32]   ComposableModel>>#gtInspectorModelNestingIn:
[32]   ComposableModel>>#gtInspectorPreviewIn:
[32]   WindowModel>>#gtInspectorModelNestingIn:
[32] 
[32] MCPackageLoader(Object)>>notify:

**Comment by jecisc** [comment]
Extensions in *Model classes should be removed. 

And classes deprecated in P7 should be removed when Migrator will be done.

**Comment by Ducasse** [comment]
The two broken tests are unrelated so I integrate!

**Pull request merged**